### PR TITLE
fix(replay): transaction replay tab should filter by proj

### PR DESCRIPTION
**Pull request by michellewzhang**
- previously in the transaction tab, we didn't pass in the project ID for our replay query, which resulted in replays from the wrong project showing up in the tab. 
- also, if no replays were related to the transaction, we were accidentally returning ALL replays for that proj rather than no replays. this was a bug introduced in https://github.com/getsentry/sentry/pull/73407
- fixes https://github.com/getsentry/team-replay/issues/486

before: query for `project:-1` and empty query string, resulting in all replays being returned.

SCR-20241015-kpcy

results in replays from a diff project:

SCR-20241015-kmsi
SCR-20241015-kmyh

after: query for a specific project based on page filters:

SCR-20241015-koil

SCR-20241015-knco
SCR-20241015-kodn

### CODE REVIEW

**Review by c298lee — APPROVED** [review]

**Comment by billyvg** [comment]
Should the replay badge be 0 when there are no replays?

File: static/app/utils/replays/fetchReplayList.tsx [hunk 1]
```diff
@@ -60,9 +63,11 @@ async function fetchReplayList({
         // we also require a project param otherwise we won't yield results
         queryReferrer,
         project:
-          queryReferrer === 'issueReplays' || queryReferrer === 'transactionReplays'
+          queryReferrer === 'issueReplays'
             ? ALL_ACCESS_PROJECTS
```
Comment (aliu39) [inline static/app/utils/replays/fetchReplayList.tsx:67]: Why do we hard code all projects for issues?

**Review by aliu39 — APPROVED** [review]

**Comment by michellewzhang** [comment]
attempt to resolve a bug introduced in https://github.com/getsentry/sentry/pull/73407 where instead of querying for `id:[]`, we accidentally query for nothing which returns all replays (not what we want) instead of no replays (what we want)

**Review by aliu39 — APPROVED** [review]

File: static/app/views/performance/transactionSummary/transactionReplays/transactionReplays.tsx [hunk 2]
```diff
@@ -137,13 +137,22 @@ function ReplaysContent({
   organization: Organization;
   pageLinks: string | null;
 }) {
-  const location = useMemo(() => ({query: {}}) as Location<ReplayListLocationQuery>, []);
+  const location = useLocation();
+
+  if (!eventView.query) {
+    eventView.query = String(location.query.query ?? '');
+  }
```
Comment (michellewzhang) [inline static/app/views/performance/transactionSummary/transactionReplays/transactionReplays.tsx:144]: @ryan953 this line fixes the test

Comment (michellewzhang) [reply static/app/views/performance/transactionSummary/transactionReplays/transactionReplays.tsx:144]: it does nothing in practice, just satisfies the test😭

Comment (ryan953) [reply static/app/views/performance/transactionSummary/transactionReplays/transactionReplays.tsx:144]: burn it all down, such a pain

**Pull request merged**